nomos-services/data-availability/indexer/src/lib.rs

```diff
@@ -523,44 +462,83 @@ where
+    DaVerifierStorage: nomos_da_verifier::storage::DaStorageAdapter<RuntimeServiceId>,
     DaVerifierBackend: nomos_da_verifier::backend::VerifierBackend + Send,
     DaVerifierBackend::Settings: Clone,
-    DaVerifierNetwork: nomos_da_verifier::network::NetworkAdapter,
+    DaVerifierNetwork: nomos_da_verifier::network::NetworkAdapter<RuntimeServiceId>,
     DaVerifierNetwork::Settings: Clone,
     TimeBackend: nomos_time::backends::TimeBackend,
     TimeBackend::Settings: Clone + Send + Sync,
     ApiAdapter: nomos_da_sampling::api::ApiAdapter + Send + Sync,
+    RuntimeServiceId: Debug
+        + Sync
+        + Display
+        + Send
+        + AsServiceId<Self>
+        + AsServiceId<
+            CryptarchiaConsensus<
+                NetAdapter,
+                BlendAdapter,
+                ClPool,
+                ClPoolAdapter,
+                DaPool,
+                DaPoolAdapter,
+                TxS,
+                BS,
+                ConsensusStorage,
+                SamplingBackend,
+                SamplingNetworkAdapter,
+                SamplingRng,
+                SamplingStorage,
+                DaVerifierBackend,
+                DaVerifierNetwork,
+                DaVerifierStorage,
+                TimeBackend,
+                ApiAdapter,
+                RuntimeServiceId,
+            >,
+        > + AsServiceId<StorageService<DaStorage::Backend, RuntimeServiceId>>,
 {
     fn init(
-        service_state: OpaqueServiceStateHandle<Self>,
+        service_state: OpaqueServiceStateHandle<Self, RuntimeServiceId>,
         _init_state: Self::State,
     ) -> Result<Self, DynError> {
-        let consensus_relay = service_state.overwatch_handle.relay();
-        let storage_relay = service_state.overwatch_handle.relay();
-
-        Ok(Self {
-            service_state,
-            storage_relay,
-            consensus_relay,
-        })
+        Ok(Self { service_state })
     }
 
     async fn run(self) -> Result<(), DynError> {
-        let Self {
-            mut service_state,
-            consensus_relay,
-            storage_relay,
-        } = self;
+        let Self { mut service_state } = self;
 
-        let consensus_relay = consensus_relay
-            .connect()
+        let consensus_relay = service_state
+            .overwatch_handle
+            .relay::<CryptarchiaConsensus<
+                NetAdapter,
+                BlendAdapter,
+                ClPool,
+                ClPoolAdapter,
+                DaPool,
+                DaPoolAdapter,
+                TxS,
+                BS,
+                ConsensusStorage,
+                SamplingBackend,
+                SamplingNetworkAdapter,
+                SamplingRng,
+                SamplingStorage,
+                DaVerifierBackend,
+                DaVerifierNetwork,
+                DaVerifierStorage,
+                TimeBackend,
+                ApiAdapter,
+                RuntimeServiceId,
+            >>()
```

Comment:
Can't the generics be elided? :thinking:

Reply:
Yeah good point. I think for some it's not possible, but for other it should be. I will do that 👍

Reply:
This is the best we can do. We can't completely remove them where there's at least two `AsServiceId` bounds: https://github.com/logos-co/nomos/pull/1183/commits/ed3f51f427c1c41a06f91d4d29746d862bedf47a.

Reply:
> This is the best we can do. We can't completely remove them where there's at least two `AsServiceId` bounds: [ed3f51f](https://github.com/logos-co/nomos/commit/ed3f51f427c1c41a06f91d4d29746d862bedf47a).

This look much better IMO. Thanks!